javascript/javascript1/week3/yasi/Smart-ease3.js

```diff
@@ -0,0 +1,205 @@
+    travelInformation.destinationDistance - days * 24 * travelInformation.speed;
+  const time = remainingDistance / travelInformation.speed;
+  console.log(time);
+  const hours = parseInt(time, 10);
+  const minutes = Math.round((time - hours) * 60);
+  return `${days} days, ${hours} hours and ${minutes} minutes`;
+}
+const travelTime = calculateTravelTime(travelInformation);
+console.log("Travel time is: ", travelTime); // 8 hours and 38 minutes*/
+
+// Series duration of my life
+const seriesDurations = [
+  {
+    title: "Game of thrones",
+    days: 3,
+    hours: 1,
+    minutes: 0,
+  },
+  {
+    title: "Sopranos",
+    days: 3,
+    hours: 14,
+    minutes: 0,
+  },
+  {
+    title: "The Wire",
+    days: 2,
+    hours: 12,
+    minutes: 0,
+  },
+];
+function logOutSeriesText() {
+  const totalLifeSpan = 80 * 365 * 24 * 60;
+  let totalSeriesTime = 0;
+
+  for (let i = 0; i < seriesDurations.length; i++) {
+    const seriesTime =
+      seriesDurations[i].days * 24 * 60 +
+      seriesDurations[i].hours * 60 +
+      seriesDurations[i].minutes;
+
+    const seriesPercentage = (seriesTime * 100) / totalLifeSpan;
+    totalSeriesTime += seriesTime;
+    console.log(
+      `${seriesDurations[i].title} took ${seriesPercentage.toFixed(
+        3
+      )}% of my life`
+    );
+  }
+  const totalPercentage = (totalSeriesTime / totalLifeSpan) * 100;
+  console.log(`\nIn total that is ${totalPercentage.toFixed(3)}% of my life`);
+}
+
+logOutSeriesText(); // logs out the text found above
+
+//NOnoN0nOYes (Note taking app)
+const notes = [];
+
+function saveNote(content, id) {
+  notes.push({ content, id });
+}
+function getNote(id) {
+  if (typeof id !== "number") {
+    console.error("Error: ID must be a number");
+    return;
+  }
+
+  for (let i = 0; i < notes.length; i++) {
+    if (notes[i].id === id) {
+      return notes[i];
+    }
+  }
+  return "Note not found";
+}
+function logOutNotesFormatted() {
+  for (let i = 0; i < notes.length; i++) {
+    console.log(
+      `The note with id: ${notes[i].id}, has the following note text: ${notes[i].content}`
+    );
+  }
+}
+
+//Unique feature
+function deleteNote(id) {
+  for (let i = 0; i < notes.length; i++) {
+    if (notes[i].id === id) {
+      notes.splice(i, 1); // Remove the note from the array
+      console.log(`Note with id ${id} deleted.`);
+      return;
+    }
+  }
+  console.log("Note not found.");
+}
+
+//Test the note taking app
+saveNote("Pick up groceries", 1);
+saveNote("Do laundry", 2);
+
+console.log(notes);
+const firstNote = getNote(1);
+console.log(firstNote);
+
+logOutNotesFormatted();
+
+deleteNote(1);
+logOutNotesFormatted();
+
+//CactusIO-interactive (Smart phone usage app)
+const activities = [];
+const usageLimit = 100;
+
+function addActivity(activity, duration) {
+  let today = new Date().toLocaleDateString("en-US");
+  activities.push({ date: today, activity, duration });
+}
+
+function showStatus() {
+  if (activities.length === 0) {
```

Comment:
Can also be written as 
```suggestion
  if (!activities.length) {
```